packages/viz/src/lib/basemap.ts

```diff
@@ -0,0 +1,112 @@
+
+/**
+ * Mapping libraries are meant to be available in the browser before using these helpers
+ */
+declare global {
+  interface Window {
+    deck: {
+      DeckGL(params?: any): any;
+      GoogleMapsOverlay(params?: any): any;
+    };
+
+    google: {
+      maps: {
+        Map(mapContainer?: any, opts?: any): any;
+      }
+    };
+  }
+}
+
+interface StyleUrlCatalog { [key: string]: string; }
+const cartoBasemaps: StyleUrlCatalog = {
+  positron: 'https://basemaps.cartocdn.com/gl/positron-gl-style/style.json',
+  voyager: 'https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json',
+  darkmatter: 'https://basemaps.cartocdn.com/gl/dark-matter-gl-style/style.json',
+};
+
+/**
+ * A helper function to create a CARTO basemap on a 'map' DOM element, rendered using *Mapbox GL JS*
+ *
+ * Examples:
+ * ```javascript
+ *    // Several options to create the map are allowed
+ *    const deckMap = carto.viz.map();
+ *    const deckMap = carto.viz.map('voyager');
+ *    const deckMap = carto.viz.map('darkmatter', { zoom: 4 });
+ *    const deckMap = carto.viz.map('positron', { zoom: 4, longitude: 3, latitude: 40, pitch: 45, bearing: 30 }, 'map');
+ * ```
+ * @export
+ * @param {string} [basemap='positron']
+ * @param {*} [view]
+ * @param {string} [containerId='map']
+ * @returns
+ */
+export function map(basemap: string = 'positron', view?: any, containerId: string = 'map') {
+  if (!window.deck.DeckGL) {
+    throw new Error(
+      'This helper is meant to be used on the browser, with mapbox & deck available at window'
+    );
+  }
+
+  const mapStyle = cartoBasemaps[basemap.toLowerCase()];
+
+  const DEFAULT_VIEW = {
+    longitude: 0,
+    latitude: 0,
+    zoom: 1,
+  };
+  const initialViewState = Object.assign({}, DEFAULT_VIEW, view);
+
+  const deckMap = new (window.deck.DeckGL as any)({
```

Comment:
We should not need to set Deck.gl as any if we had previously defined types in TS

Reply:
The type defines an interface but it doesn't define a ctor function... I think we have to keep it by now, or change the window declaration